### PR TITLE
Move replicaton task filter to stream sender

### PR DESCRIPTION
**Pull request by yux0**
## What changed?
Move replicaton task filter to stream sender

## Why?
Task convertor should just do conversion.

### CODE REVIEW

**Review by xwduan — APPROVED** [review]

File: service/history/replication/raw_task_converter.go [hunk 1]
```diff
@@ -52,77 +52,44 @@ import (
...
-	}
-
-	if !shouldProcessTask {
-		return nil, nil
-	}
-
-	clientShardID := common.WorkflowIDToHistoryShard(task.GetNamespaceID(), task.GetWorkflowID(), c.clientClusterShardCount)
-	if clientShardID != c.clientShardKey.ShardID {
-		return nil, nil
+		ctx, cancel = context.WithTimeout(context.Background(), applyReplicationTimeout)
```
Comment (yycptt) [inline service/history/replication/raw_task_converter.go:89]: mark the context as lowest priority?

Comment (yycptt) [reply service/history/replication/raw_task_converter.go:89]: actually, just use empty string as nsName and fall through to `newTaskContext` below?

`if there is error, then blindly send the task, better safe than sorry`
^ comment can also be removed I think.

File: service/history/replication/fx.go [hunk 2]
```diff
@@ -107,16 +106,11 @@ func replicationTaskConverterFactoryProvider() SourceTaskConverterProvider {
 	return func(
 		historyEngine shard.Engine,
 		shardContext shard.Context,
-		clientClusterShardCount int32,
 		clientClusterName string,
```
[the remark this answers is not in the document]
Comment (yux0) [reply service/history/replication/fx.go:109]: I added a comment: clientClusterName string, // Task converter is cluster aware.

[the remark this answers is not in the document]
Comment (yycptt) [reply service/history/replication/fx.go:109]: probably should explicitly call out certain converter implementation depends on the field.

**Review by yycptt — APPROVED** [review]

**Pull request merged**